app/js/components/searchHistory/searchHistoryComponent.jsx

```diff
@@ -57,8 +57,8 @@ class SearchHistoryComponent extends Component {
     viewResult(index) {
         const parameters = (typeof index === 'number')? this.state.searchHistory[index].parameters : index
         this.search(parameters).then((results) => {
-            const allPatients = results.rows;
-            const pagePatientInfo = this.getPagePatient(allPatients, this.state.currentPage);
+            const allPatients = results.rows || [];
+            const pagePatientInfo = this.getPagePatient(allPatients, 1);
```

Comment:
The method that encapsulates this code is only called when a user decides to view a result, the result has been divided for easy pagination. The previous code sets the current page to display from the state. This resulted to a bug because the previous search might have changed the value of currentPage to 5. In this instance, I only want to see the first paginated data.